app/Main.hs

```diff
@@ -129,5 +42,25 @@ showCols cols = do
   putStrLn "---------------"
   showCols cols
 
+randomNums :: Int -> IO String
+randomNums x = do
+  g <- newStdGen
+  return $ take x (randomRs ('1', '9') g)
+
+numberWang :: IO ()
+numberWang = do
+  num <- randomNums 1
+  print num
+  attempt <- getLine
+  if attempt == snd (baseNumbers !! (read num - 1))
+    then
+    do
+      putStrLn "That's NUMBERWANG!"
+      numberWang
```

Comment:
Nit: Since you execute `numberWang` in both of the branches, you can put it outside the if/else